Add Kinetis USB Phy

c1728p9 opened the pull request:
- Add Kinetis USBPhy
- Add extra checks to USBDevice
- Fix read_start behavior when configuring

c1728p9 commented:
Please review @maciejbocianski and @fkjagodzinski

maciejbocianski commented on targets/TARGET_Freescale/usb/USBPhy_Kinetis.cpp:
```diff
@@ -0,0 +1,638 @@
+    NVIC_EnableIRQ(USB0_IRQn);
+}
+
+void USBPhyHw::deinit()
```
shouldn't we add disconnecting code here?
similar to LPC176X implementation

c1728p9 replied:
Good catch. It has been added.

c1728p9 commented:
@ashok-rao the [USB device HAL specification](https://github.com/ARMmbed/Handbook/blob/new_engine/docs/reference/contributing/target/usb.md#testing) explains how to run the tests:
```
mbed test -t <toolchain> -m <target> -n mbed-os-tests-usb_device-*
```

c1728p9 commented:
Addressed @maciejbocianski's comment and also simplified the USBDevice read API (user API not HAL API). The older API allowed reads to be explicitly started, which some hardware doesn't support (Like the LPC1768). With the new API the receiver is automatically started when the endpoint is added.

Old endpoint state machine diagram:

![usb_endpoint_state_diagram_user](https://user-images.githubusercontent.com/2916772/37437097-51f9837a-27b9-11e8-8985-0a1f6210209e.png)

New endpoint state machine diagram:

![usb_endpoint_state_diagram_user](https://user-images.githubusercontent.com/2916772/37437016-e844ded4-27b8-11e8-8c48-58f320f74cfc.png)

The [USB technology docs PR](https://github.com/ARMmbed/Handbook/pull/440) has also been updated to reflect this.

cmonr commented:
/morph build

cmonr commented:
Will relaunch when able to. ARM license CI issue.

cmonr commented:
/morph build

cmonr commented:
CI issue: NRF52 flash cache test

Rebuilding due to new PR commits.
/morph build

cmonr commented:
CI issue: NRF52 flash cache test

Restarting.
/morph test
